tests/structured_conf/test_structured_config.py

```diff
@@ -864,6 +864,135 @@ def test_create_untyped_list(self, class_type: str) -> None:
         assert cfg.list == [1, 2]
         assert cfg.opt_list is None
 
+    class TestInstantiateStructuredConfigs:
+        @pytest.fixture
+        def module(self, class_type: str) -> Any:
+            module: Any = import_module(class_type)
+            return module
+
+        def round_trip_to_object(self, input_data: Any) -> Any:
+            serialized = OmegaConf.create(input_data)
+            round_tripped = OmegaConf.to_object(serialized)
+            return round_tripped
+
+        def test_basic(self, module: Any) -> None:
+            user = self.round_trip_to_object(module.User("Bond", 7))
+            assert isinstance(user, module.User)
+            assert type(user) is module.User
+            assert user.name == "Bond"
+            assert user.age == 7
+
+        def test_basic_with_missing(self, module: Any) -> None:
+            user = self.round_trip_to_object(module.User())
+            assert isinstance(user, module.User)
+            assert type(user) is module.User
+            assert user.name == MISSING
+            assert user.age == MISSING
+
+        def test_nested(self, module: Any) -> None:
```

Comment:
When I said that I don't thin you are handling nesting, you said that you were.
looking at this test, the nesting you were thinking about is a Structured Config nested in a dict/list.
I was thinking about a Structured Config nested in another Structured Config.

e.g: a Plugin inside a PluginHolder.

Reply:
This case is handled by the `test_nested_object` test defined further down in the file:
```python3
        def test_nested_object(self, module: Any) -> None:
            nested = self.round_trip_to_object(module.NestedConfig)
            assert isinstance(nested, module.NestedConfig)
            assert type(nested) is module.NestedConfig

            assert nested.default_value == MISSING

            assert isinstance(nested.user_provided_default, module.Nested)
            assert type(nested.user_provided_default) is module.Nested
            assert nested.user_provided_default.with_default == 42
```

Reply:
Okay, cool.